src/org/sosy_lab/java_smt/solvers/z3/Z3TheoremProver.java

```diff
@@ -137,8 +137,12 @@ protected long getUnsatCore0() {
   }
 
   @Override
-  protected long getZ3Model() {
-    return Native.solverGetModel(z3context, z3solver);
+  protected long getZ3Model() throws InterruptedException {
+    try {
+      return Native.solverGetModel(z3context, z3solver);
+    } catch (Z3Exception e) {
+      throw new InterruptedException("Z3 model generation interrupted.");
```

Comment:
This code is wrong. Z3Exceptions can have other causes as well. Please use proper handling from Z3FormulaCreator.handleZ3Exception(e).

Reply:
fixed